src/debug.cpp

```diff
@@ -247,18 +247,59 @@ std::ostream &operator<<( std::ostream &out, DebugClass cl )
     return out;
 }
 
+struct time_info {
+    int hours;
+    int minutes;
+    int seconds;
+    int mseconds;
+
+    template <typename Stream>
+    friend Stream& operator<<(Stream& out, time_info const& t) {
+        using char_t = typename Stream::char_type;
+        using base   = std::basic_ostream<char_t>;
+
+        static_assert(std::is_base_of<base, Stream>::value, "");
+
+        out << t.hours   << ':' << t.minutes << ':'
+            << t.seconds << '.' << t.mseconds;
+
+        return out;
+    }
+};
+
+#ifdef _MSC_VER 
+inline time_info get_time() noexcept {
+    SYSTEMTIME time {};
+
+    GetLocalTime(&time);
+
+    return time_info {
+        static_cast<int>(time.wHour)
+      , static_cast<int>(time.wMinute)
+      , static_cast<int>(time.wSecond)
+      , static_cast<int>(time.wMilliseconds)
+    };
+}
+#else
+inline time_info get_time() noexcept {
+    timeval tv {};
+    gettimeofday( &tv, nullptr );
+
+    auto const tt      = tv.tv_sec;
+    auto const current = localtime( &tt );
+
+    return time_info {
+        current->tm_hour
+      , current->tm_min
+      , current->tm_sec
+      , static_cast<int>(tv.tv_usec / 1000 + 0.5)
```

Comment:
I took the code as written -- looking at it again, it's a bit odd isn't it;

```
static_cast<int>(tv.tv_usec / 1000.0 + 0.5) would be more sane
```

I'll change it